lang_GENERIC/analyze/Dataflow_constness.ml

```diff
@@ -0,0 +1,286 @@
+(* Iago Abal
+ *
+ * Copyright (C) 2020 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file license.txt.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * license.txt for more details.
+*)
+open Common
+open IL
+
+module G = AST_generic
+module F = IL
+module D = Dataflow
+module VarMap = Dataflow.VarMap
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+(* The type of an unknown constant. *)
+type ctype =
+  | Cbool
+  | Cint
+  | Cstr
+  | Cany
+
+type constness = Lit of AST_generic.literal | Cst of ctype | Dyn
```

Comment:
I think you can put this in AST_generic.ml.
Maybe instead of Dyn you can put NotAConst
or use an option type for the whole thing.
then you can rename AST_generic.id_info.id_const_literal to id_constness using this type in a ref
so we don't need the <arbitrary_string> hack.

Reply:
Sure, then I'll add a new kind of _arbitrary_ `literal`.